CustomDate - copia.js

```diff
@@ -0,0 +1,127 @@
+class CustomDate {
+    constructor() {
+        this.now = new Date();
+        this.date = new Date();
+        
+        //Use constants to avoid 'magic numbers'
+        this.daysOfTheWeek = ['SUNDAY', 'MONDAY', 'TUESDAY', 'WEDNESDAY', 'THURSDAY', 'FRIDAY', 'SATURDAY'];
+        this.HoursInDay = 24;
+        this.MinsInHour = 60;
+        this.SecondsInMin = 60;
+        this.msInSecond = 1000;
+    }
+
+
+    // update this.date object
+    updateDate(newDate) {
+        this.date = newDate;
+    }
+
+
+    // return yesterday's date
+    yesterday() {
+        const yesterday = new Date();
+        yesterday.setDate(now.getDate() - 1);
+        return yesterday;
+    }
+
+
+    // return yesterday's day
+    yesterdayDay() {
+        return this.daysOfTheWeek[(this.now.getDay() - 1) % 7];
+    }
+
+
+    // return tomorrow's date
+    tomorrow() {
+        const tomorrow = new Date();
+        tomorrow.setDate(now.getDate() + 1);
+        return tomorrow;
+    }
+
+
+    // return tomorrow's day
+    tomorrowDay() {
+        return this.daysOfTheWeek[(this.now.getDay() + 1) % 7];
+    }
+
+
+    // add minutes to this.date
+    addMinutes(minutes) {
+        this.date.setMinutes(this.date.getMinutes() + minutes);
+    }
+
+
+    // add hours to this.date
+    addHours(hours) {
+        this.date.setHours(this.date.getHours() + hours);
+    }
+
+
+    // add days to this.date
+    addDays(days) {
+        this.date.setDate(this.date.getDate() + days);
+    }
+
+
+    // Logs the current date-time
+    current() {
+        console.log(this.now);
+    }
+
+
+    // return boolean
+    isFuture() {
+        return this.now < this.date;
+    }
+
+
+    // Return (bigger, smaller or equal) between the given date and this.date
+    compare(newDate) {
+        if (newDate > this.date) {
+            return 'Bigger';
+        } else if (this.date) {
+            return 'Smaller'
+        } else {
+            return 'Equal';
+        }
+    }
+
+
+    // Return the difference between the given date and this.date in format (x Days, y hours, z minutes)
+    differenceInDays(newDate) {
+        const difference = Math.abs(this.date - newDate);
```

Comment:
can you check this again
![image](https://github.com/faresrauf/Task2-and-Modules-Project/assets/87116208/649c95d9-c75e-4122-823f-251cfde862d2)

Reply:
I run it using date.differenceInDays(new Date(2024,7,8));
And it worked giving the difference in days, by converting to ms and dividing by the total numbar of milliseconds in one day.